src/CloudFtpBridge.Core/Services/WorkflowRunner.cs

```diff
@@ -196,5 +200,54 @@ public async Task<bool> Run(Workflow workflow)
 
             return !hasErrors && hasFiles;
         }
+
+        private string GetReference(string fileName, Stream inData, string regexOrXpath)

```

Comment:
This should probably be put in a separate service instead of in the workflow runner, but I'll just log this as tech debt for a future release. It isn't urgent.